src/libraries/System.Private.CoreLib/src/System/Reflection/Emit/ModuleBuilder.cs

```diff
@@ -80,7 +80,11 @@ public TypeBuilder DefineType(string name, TypeAttributes attr)
 
         public TypeBuilder DefineType(string name, TypeAttributes attr,
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.All)] Type? parent, Type[]? interfaces)
-                => DefineTypeCore(name, attr, parent, interfaces, PackingSize.Unspecified, TypeBuilder.UnspecifiedTypeSize);
+        {
+            ArgumentException.ThrowIfNullOrEmpty(name);
```

Comment:
Als delete these arg validations from the `Core*` methods in Runtime-specific ModuleBuilders?